### PR TITLE
[PWA-1934] PoC: Banner and Slider Optimizations

### PR DESCRIPTION
**Pull request by mikhaelbois**
## Description

Improve loading experience for Page Builder Banner and Slider components.

- Create Shimmer components for Banner and Slider components
- Implement Shimmer logic in Page Builder Factory so other Shimmer components can be created
- Add lazy loading in Slick Slider for the Slider and Product Carousel components
- Load Banner background image only if visible in viewport

Note: Banner and Slider components needs a minimum height for the Shimmer to be useful.

## Related Issue

Proof of Concept for https://jira.corp.magento.com/browse/PWA-1934
Not linked to any opened Github issue.

### Verification Stakeholders

@larsroettig @supernova-at @revanth0212 

#### Test scenario(s) for direct fix/feature

You'll need access to an admin so you can play around Page Builder elements.

1. Create a Slider component in a CMS Page.
2. Test with and without minimum height for Slider and Slides components.
3. Test shimmer effect.

1. Create a Banner component in a CMS Page.
2. Test with and without minimum height.
3. Test shimmer effect.

1. Create a Product Carousel component in a CMS Page.
2. Test carousel lazy load.

#### Is Browser/Device testing needed?

Yes, browser and device testing is needed to test regression on mobile / desktop and all major browsers.

#### Any ad-hoc/edge case scenarios that need to be considered?

Now that the slides and background images are lazy loaded, we need to make sure the user experience is not too impacted.
Having no minimum height is not ideal and will impact the experience.

## Checklist

- [x] I have added tests to cover my changes.

### CODE REVIEW

File: packages/venia-ui/lib/components/HomePage/homePage.css [hunk 1]
```diff
@@ -80,6 +80,7 @@
  * Target the slider.
  */
 :global(.venia-home-slider .slick-slider) {
+    min-height: 628px !important;
```
Comment (michaelyu0123) [inline packages/venia-ui/lib/components/HomePage/homePage.css:83]: How are these min-height decided, and can we work with device height instead?

**Review by michaelyu0123 — CHANGES_REQUESTED** [review]

File: packages/venia-ui/lib/components/HomePage/homePage.css (hunk 1 above)
Comment (mikhaelbois) [reply packages/venia-ui/lib/components/HomePage/homePage.css:83]: This is the height of the content with the height of the pagination.
Since I couldn't update the min-height from the backend, CSS was the only way to apply it for the mobile and desktop versions and avoid a div without any height.

File: packages/pagebuilder/lib/ContentTypes/Banner/__tests__/__snapshots__/banner.spec.js.snap [hunk 2]
```diff
@@ -171,7 +177,7 @@ exports[`renders a configured collage-left Banner component 1`] = `
         Object {
           "backgroundAttachment": "scroll",
           "backgroundColor": "blue",
-          "backgroundImage": "url(desktop-image.jpg)",
+          "backgroundImage": "url(null)",
```
[the remark this answers is not in the document]
Comment (mikhaelbois) [reply packages/pagebuilder/lib/ContentTypes/Banner/__tests__/__snapshots__/banner.spec.js.snap:180]: I can check to update the test yes.

**Review by michaelyu0123 — APPROVED** [review]

**Comment by michaelyu0123** [comment]
LGTM

**Comment by pankhurigoel** [comment]
> Fails
> 🚫	
> node` failed.
> 
> 🚫	
> The following file(s) were not formatted with **prettier**. Make sure to execute `yarn run prettier` locally prior to committing.
> 
> ```
> packages/venia-ui/lib/components/MegaMenu/megaMenuItem.js
> ```
> 
> Messages
> 📖	
> Associated JIRA tickets: [PWA-1934](https://jira.corp.magento.com/browse/PWA-1934).
> 
> 📖	DangerCI Failures related to missing labels/description/linked issues/etc will persist until the next push or next nightly build run (assuming they are fixed).
> 📖	
> Access a deployed version of this PR [here](https://pr-3328.pwa-venia.com/). Make sure to wait for the "pwa-pull-request-deploy" job to complete.
> 
> ### Log
> ERROR ON TASK: prettierCheck
> 
> ```
> 
> Error:  Danger had errors running. See message(s) above for more details.
> danger-results://tmp/danger-results.json
> ```
> 
> Generated by 🚫 [dangerJS](https://danger.systems/js) against [2b1a2a7](https://github.com/magento/pwa-studio/commit/2b1a2a7c82abcafc7f7c911492e8c5085d5bfabd)

@mikhaelbois please fix prettier errors.

**Comment by mikhaelbois** [comment]
@pankhurigoel I fixed the prettier issue.
